docs/developers/Custom_Rules.mdx

```diff
@@ -121,6 +121,58 @@ export const rule = ESLintUtils.RuleCreator.withoutDocs({
 We recommend any custom ESLint rule include a descriptive error message and link to informative documentation.
 :::
 
+## Handling rule options
+
+ESLint rules can take options. When handling options, you will need to add information in at most three places:
+
+- The `Options` generic type argument to `RuleCreator`, where you declare the type of the options
+- The `meta.schema` property, where you add a JSON schema describing the options shape
+- The `defaultOptions` property, where you add the default options value
+
+```ts
+type MessageIds = 'lowercase' | 'uppercase';
+
+type Options = [
+  {
+    preferredCase?: 'lower' | 'upper';
+  },
+];
```

Comment:
I have one question: should this be the type _before_ or _after_ validation (and applying defaults)? My guess is "after", but I don't know if it's used elsewhere than checking `defaultOptions` and typing the `options` parameter.